src/ng/filter/filter.js

```diff
@@ -196,7 +196,7 @@ function deepCompare(actual, expected, comparator, matchAgainstAnyProp) {
       } else if (expectedType === 'object') {
         for (key in expected) {
           var expectedVal = expected[key];
-          if (isFunction(expectedVal)) {
+          if (isFunction(expectedVal) || isUndefined(expectedVal)) {
```

Comment:
It was successfully landed to `1.3.15`, look at the [changelog](https://github.com/angular/angular.js/blob/master/CHANGELOG.md#1315-locality-filtration-2015-03-17)